L1Trigger/L1TNtuples/src/L1AnalysisRecoMuon2.cc

```diff
@@ -92,6 +92,6 @@ void L1Analysis::L1AnalysisRecoMuon2::SetMuon(const edm::Event& event,
 }
 
 void L1Analysis::L1AnalysisRecoMuon2::init(const edm::EventSetup& eventSetup) {
-  muPropagator1st_.init(eventSetup);
-  muPropagator2nd_.init(eventSetup);
+  muPropagator1st_ = muPropagatorInt1st_.init(eventSetup);
+  muPropagator2nd_ = muPropagatorInt2nd_.init(eventSetup);
```

Comment:
It's not obvious to me that this is thread-safe, as it modifies class members whenever `init` is called (which is per-event, I guess). On the other hand, looks like the only plugin using `L1AnalysisRecoMuon2` is a `one::EDAnalyzer` (i.e. `L1Muon2RecoTreeProducer`), so this might be a non-issue (as long as users of `L1AnalysisRecoMuon2` know that it may not be thread-safe?).

Reply:
Yes, that is a bit on issue here. Basically, `L1AnalysisRecoMun2` has the same issue with thread safety that `PropagateToMuon` used to have. As you said, this is probably fine in the its one use case. One could probably refactor this class the same way as `PropagateToMuon`, but as this does not concern the muon HLT, I do not plan on doing so.